docs/user-guide/building-assistants.rst

```diff
@@ -0,0 +1,1018 @@
+
+The intents and entities will need to be added to your domain as well:
+
+.. code-block:: yaml
+
+   intents:
+     - greet
+     - bye
+     - thank
+     - faq
+     - contact_sales
+     - inform
+
+   entities:
+     - company
+     - job_function
+     - person_name
+     - budget
+     - business_email
+     - use_case
+
+A story for a form is very simple, as all the slot collection form happens inside the form, and
+therefore doesn’t need to be covered in your stories.
+
+.. code-block:: md
+
+   ## sales form
+   * contact_sales
+       - sales_form
+       - form{"name": "sales_form"}
+       - form{"name": null}
+
+As a final step, let’s add the FormPolicy to our config file:
+
+.. code-block:: yaml
+
+   policies:
+     - name: MemoizationPolicy
+     - name: KerasPolicy
+     - name: MappingPolicy
+     - name: FormPolicy
+
+At this point, you already have a working form, so let’s try it out. Make sure to uncomment the
+``action_endpoint`` in your ``endpoints.yml`` to make Rasa aware of the action server that will run our form:
+
+.. code-block:: yaml
+
+   action_endpoint:
+    url: "http://localhost:5055/webhook"
+
+Then start the action server in a new terminal window:
+rasa run actions
+
+Then you can retrain and talk to your bot:
+
+.. code-block:: bash
+
+   rasa train
+   rasa shell
+
+This simple form will work out of the box, however you will likely want to add a bit
+more capability to handle different situations. One example of this is validating
+slots, to make sure the user provided information correctly (read more about it
+`here <https://rasa.com/docs/rasa/core/forms/#validating-user-input>`__). 
+
+Another example is that you may want to fill slots from things other than entities
+of the same name. E.g. for the "use case" situation in our Form, we would expect
+the user to type a full sentence and not something that you could necessarily
+extract as an entity. In this case we can make use of the ``slot_mappings`` method,
+where you can describe what your entities should be extracted from. Here we can
+use the ``from_text`` method to extract the users whole message:
+
+.. code-block:: python
+
+    def slot_mappings(self):
+        # type: () -> Dict[Text: Union[Dict, List[Dict]]]
+        """A dictionary to map required slots to
+        - an extracted entity
+        - intent: value pairs
+        - a whole message
+        or a list of them, where a first match will be picked"""
+        return {"use_case": self.from_text(intent="inform")}
+
+Now our bot will extract the full user message when asking for the use case slot,
+and we don’t need to use the ``use_case`` entity defined before.
+
+All of the methods within a form can be customised to handle different branches in your
+business logic. Read more about this `here <https://rasa.com/docs/rasa/core/forms/#>`__.
+However, you should make sure not to handle any unhappy paths inside the form.
```

Comment:
these should be handled by writing regular stories so your model can handle the behaviour